test/bats/test.bats

```diff
@@ -363,3 +363,37 @@ EOF
   kubectl delete --ignore-not-found deploy error-deployment valid-deployment system-error-deployment
   kubectl delete --ignore-not-found constrainttemplate k8sexternaldata
 }
+
+@test "gatekeeper expansion test" {
+  # setup ns, TemplateExpansion and Constraints
+  run kubectl create namespace loadbalancers
+  run kubectl apply -f test/expansion/expand_deployments.yaml
+  assert_success
+  run kubectl apply -f test/expansion/k8srequiredlabels_ct.yaml
+  run kubectl apply -f test/expansion/loadbalancers_must_have_env.yaml
+  wait_for_process ${WAIT_TIME} ${SLEEP_TIME} "constraint_enforced k8srequiredlabels loadbalancers-must-have-env"
+
+  # assert that creating deployment without 'env' label is rejected
+  run kubectl apply -f test/expansion/deployment_no_label.yaml
+  assert_failure
+  # a deployment with the required label should succeed
+  run kubectl apply -f test/expansion/deployment_with_label.yaml
+  assert_success
+  run kubectl delete -f test/expansion/deployment_with_label.yaml
+
+  # create deployment without 'env' label and assignmetadata to add 'env'
+  run kubectl apply -f test/expansion/assignmeta_env.yaml
+  # wait for mutation to be registered by controllers
+  wait_for_process ${WAIT_TIME} ${SLEEP_TIME} "sleep 5"
+  run kubectl apply -f test/expansion/deployment_no_label.yaml
+  assert_success
+
```

Comment:
what is the expectation here? Deployment gets mutated with `env` label? If so, can we add a check for that?

Reply:
The expectation is as follows:

* In the prior request, the same deployment was rejected because it would have created pods that lack the required label
* Now we have a mutator that would add the required pod label, as a result the previously rejected request should work
* A subsequent kubectl apply should recognize that the mutation is honored and be met with success.

Added a comment to clarify.